safety_shield/tests/src/robot_reach_tests.cc

```diff
@@ -274,6 +275,78 @@ TEST_F(RobotReachTest, ApproximateVelOfCapsuleTest1) {
   EXPECT_NEAR(v_approx, 0.11, 1e-8);
 }
 
+// jacobian of tcp from siciliano p. 114
+// a1 = a2 = a3 = 1
+Eigen::Matrix<double, 6, Eigen::Dynamic> jacobian_tcp_siciliano(std::vector<double> q) {
+    Eigen::Matrix<double, 6, Eigen::Dynamic> jacobian;
+    jacobian.setZero(6, 3);
+    double element_00 = -sin(q[0]) - sin(q[0] + q[1]) - sin(q[0] + q[1] + q[2]);
+    double element_01 = cos(q[0]) + cos(q[0] + q[1]) + cos(q[0] + q[1] + q[2]);
+    double element_10 = -sin(q[0] + q[1]) - sin(q[0] + q[1] + q[2]);
+    double element_11 = cos(q[0] + q[1]) + cos(q[0] + q[1] + q[2]);
+    double element_20 = -sin(q[0] + q[1] + q[2]);
+    double element_21 = cos(q[0] + q[1] + q[2]);
+
+    Eigen::Vector<double, 6> col_0;
+    col_0 << element_00, element_01, 0, 0, 0, 1;
+    jacobian.col(0) = col_0;
+
+    Eigen::Vector<double, 6> col_1;
+    col_1 << element_10, element_11, 0, 0, 0, 1;
+    jacobian.col(1) = col_1;
+
+    Eigen::Vector<double, 6> col_2;
+    col_2 << element_20, element_21, 0, 0, 0, 1;
+    jacobian.col(2) = col_2;
+
+    return jacobian;
+}
+
+TEST_F(RobotReachTestVelocity, JacobianSicilianoTest0) {
+    std::vector<double> q = {0.0, 0.0, 0.0};
+    robot_reach_->calculateAllTransformationMatricesAndCapsules(q);
+    reach_lib::Point point = robot_reach_->getTcpOfVelocityCapsule();
+    Eigen::Matrix<double, 6, Eigen::Dynamic> jacobian = robot_reach_->getJacobian(2, point);
+    Eigen::Matrix<double, 6, Eigen::Dynamic> expect = jacobian_tcp_siciliano(q);
+    ASSERT_TRUE((jacobian - expect).norm() < 1e-8);
```

Comment:
If something fails here, you get no feedback, why. 
I would use 3*6 `EXPEXT_NEAR()` statements. 
This looks a bit ridiculous but gives you the most insights why the test failed.
You could maybe write a function for these 18 EXPECTS to reuse them later.